src/messaging/ExchangeContext.cpp

```diff
@@ -129,11 +138,12 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
     const Transport::PeerAddress * peerAddress = GetSessionHandle().GetPeerAddress(mExchangeMgr->GetSessionManager());
     // Treat unknown peer address as "not UDP", because we have no idea whether
     // it's safe to do MRP there.
-    bool isUDPTransport                = peerAddress && peerAddress->GetTransportType() == Transport::Type::kUdp;
-    bool reliableTransmissionRequested = isUDPTransport && !sendFlags.Has(SendMessageFlags::kNoAutoRequestAck);
+    bool isUDPTransport = peerAddress && peerAddress->GetTransportType() == Transport::Type::kUdp;
+    bool reliableTransmissionRequested =
+        isUDPTransport && !sendFlags.Has(SendMessageFlags::kNoAutoRequestAck) && !IsGroupExchangeContext();
```

Comment:
This change is not needed, because we already do `SetAutoRequestAck(!session.IsGroupSession())` and `reliableTransmissionRequested` is ignored if `!AutoRequestAck()`.

Reply:
True, I'll removed the check but add a comment because this isn't obvious